core/provider/service.go

```diff
@@ -209,6 +211,23 @@ func (s *Service) Update(ctx context.Context, p *domain.Provider) error {
 		}
 	}
 
+	go func() {
+		s.logger.Info("fetching resources", "provider_urn", p.URN)
+		ctx := audit.WithActor(context.Background(), domain.SystemActorName)
+		resources, err := s.getResources(ctx, p)
+		if err != nil {
+			s.logger.Error("failed to fetch resources", "error", err)
+		}
+
+		if !dryRun {
+			if err := s.resourceService.BulkUpsert(ctx, resources); err != nil {
+				s.logger.Error("failed to insert resources to db", "error", err)
+			} else {
+				s.logger.Info("resources added", "provider_urn", p.URN, "count", len(resources))
+			}
+		}
+	}()
```

Comment:
since it is a goroutine, do we even have to run it on a dry-run call?

Reply:
Yes. was thinking the same, but since the create provider has a similar logic, replicated it here.